core/ArchiveProcessor.php

```diff
@@ -770,13 +772,59 @@ public function processDependentArchive($plugin, $segment)
             return;
         }
 
+        // The below check is meant to avoid archiving the same dependency multiple times.
+        $processedSegmentKey = $params->getSite()->getId() . $params->getDateStart() . $params->getPeriod()->getLabel() . $newSegment->getOriginalString();
```

Comment:
```suggestion
        $processedSegmentKey = $params->getSite()->getId() . $params->getPeriod()->getDateStart() . $params->getPeriod()->getLabel() . $newSegment->getOriginalString();
```

Given we are using `getPeriod()->getDateStart()` to invalidate the correct periods, I would also use that one for the key. It may avoid confusion some future as the values are different if the site is not set up with UTC.